fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt

```diff
@@ -13,23 +18,100 @@ import org.mozilla.fenix.settings.SupportUtils
  * Returns a list of all the required Nimbus 'cards' that have been converted to [OnboardingPageUiData].
  */
 internal fun Collection<OnboardingCardData>.toPageUiData(
+    context: Context,
     showNotificationPage: Boolean,
     showAddWidgetPage: Boolean,
-): List<OnboardingPageUiData> =
-    filter {
-        when (it.cardType) {
-            OnboardingCardType.NOTIFICATION_PERMISSION -> {
-                it.enabled && showNotificationPage
-            }
-            OnboardingCardType.ADD_SEARCH_WIDGET -> {
-                it.enabled && showAddWidgetPage
-            }
-            else -> {
-                it.enabled
+): List<OnboardingPageUiData> {
+    val junoOnboardingFeature = FxNimbus.features.junoOnboarding.value()
+    val jexlConditions = junoOnboardingFeature.conditions
+
+    // we are first filtering the cards based on Nimbus configuration
+    return filter { it.shouldDisplayCard(context, jexlConditions) }
+        // we are then filtering again based on device capabilities
+        .filter {
+            when (it.cardType) {
+                OnboardingCardType.NOTIFICATION_PERMISSION -> {
+                    it.enabled && showNotificationPage
+                }
+
+                OnboardingCardType.ADD_SEARCH_WIDGET -> {
+                    it.enabled && showAddWidgetPage
+                }
+
+                else -> {
+                    it.enabled
+                }
             }
-        }
-    }.sortedBy { it.ordering }
+        }.sortedBy { it.ordering }
         .map { it.toPageUiData() }
+}
+
+/**
+ *  Returns true if the the current Nimbus card fulfills all the conditions in order to display it and false otherwise
+ *
+ *  @param context The context used to fetch the jexlHelper.
+ *  @param jexlConditions A <String, String> map containing the Nimbus conditions
+ *
+ *  @return True - if the card should be displayed and False otherwise.
+ */
+fun OnboardingCardData.shouldDisplayCard(
+    context: Context,
+    jexlConditions: Map<String, String>,
+): Boolean {
+    val jexlCache: MutableMap<String, Boolean> = mutableMapOf()
+    val jexlHelper = context.components.analytics.messagingStorage.helper
+
+    return verifyConditionEligibility(
+        prerequisites,
+        jexlConditions,
+        jexlCache,
+        jexlHelper,
+    ) && !verifyConditionEligibility(disqualifiers, jexlConditions, jexlCache, jexlHelper)
+}
+
+/**
+ * Returns true if the card's condition list is valid and false otherwise
+ *
+ * @param cardConditions The conditions (prerequisites or disqualifiers) to be verified for eligibility.
+ * @param jexlConditions The Nimbus conditions map.
+ * @param jexlCache A cache that contains the already evaluated conditions and their eligibility.
+ * @param jexlHelper A helper used to evaluate the jexl conditions.
+ *
+ * @return True - if the conditions are valid and False otherwise.
```

Comment:
nit: The description could perhaps be more descriptive (similar to the long comment in the fun?)

```suggestion
 * Evaluates the eligibility of given [cardConditions] against the [jexlConditions].
 *
 * @param cardConditions The conditions (prerequisites or disqualifiers) to be verified for eligibility.
 * @param jexlConditions The Nimbus conditions map.
 * @param jexlCache A cache that contains the already evaluated conditions and their eligibility.
 * @param jexlHelper A helper used to evaluate the JEXL conditions.
 *
 * @return True if the card's condition list is valid.
```